### PR TITLE
Adding floodsub

### PR DESCRIPTION
**Pull request by jrhea**
## PR Description
- added options for different gossip protocols
- stubbed out method for handling floodSub messages
## Fixed Issue(s)

#584

### CODE REVIEW

File: networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java [hunk 1]
```diff
@@ -42,53 +43,67 @@
...
-  private final EventBus eventBus;
-  private final String userAgent;
-  private final Peer peer;
-  private final ChainStorageClient store;
-  private final Set<Long> pendingResponses = new HashSet<>();
-  private final AtomicBoolean status = new AtomicBoolean(true);
-  private final Consumer<Bytes> messageSender;
-  private final Runnable handlerTermination;
-  private final State p2pState;
+public abstract class SocketHandler {
```
Comment (cemozerr) [inline networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java]: You might wanna change this file's name.

File: networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java [hunk 2]
```diff
@@ -42,53 +43,67 @@
...
 
-  private void closed(@Nullable Void nothing) {
+  @SuppressWarnings({"rawtypes"})
+  public static Class getSocketHandlerType(String gossipProtocol) {
+    if (gossipProtocol.equalsIgnoreCase(P2PNetwork.GossipProtocol.FLOODSUB.name())) {
+      return FloodsubSocketHandler.class;
+    } else if (gossipProtocol.equalsIgnoreCase(P2PNetwork.GossipProtocol.PLUMTREE.name())) {
+      return PlumtreeSocketHandler.class;
+    } else {
+      return PlumtreeSocketHandler.class;
```
Comment (cemozerr) [inline networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java]: Why this else block here?

File: networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java (hunk 1 above)
Comment (jrhea) [reply networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java]: changing now

File: networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java (hunk 2 above)
Comment (jrhea) [reply networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java]: why not

Comment (jrhea) [reply networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/hobbits/SocketHandler.java]: remove the else

**Review by cemozerr — APPROVED** [review]
Lgtm.

**Pull request merged**